File: Ahorn/entities/ELSTerminaBoss.jl
module ELSTerminaBoss

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ELSTerminaBoss" ELSTerminaBoss(x::Integer, y::Integer, fromCutscene::Bool=false, hardMode::Bool=false, phase::Integer=4)

const placements = Ahorn.PlacementDict(
    "default" => Ahorn.EntityPlacement(ELSTerminaBoss)
)

function Ahorn.selection(entity::ELSTerminaBoss)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ELSTerminaBoss, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/ELS_Termina_Boss.lua
local utils = require("utils")

local elsTerminaBoss = {}

elsTerminaBoss.name = "MaggyHelper/ELSTerminaBoss"
elsTerminaBoss.depth = -12500
elsTerminaBoss.placements = {
    name = "default",
    data = {
        phase = 4,
        fromCutscene = false,
        hardMode = false
    }
}

elsTerminaBoss.fieldInformation = {
    phase = {
        fieldType = "integer",
        options = {1, 2, 3, 4},
        editable = true
    },
    fromCutscene = {
        fieldType = "boolean"
    },
    hardMode = {
        fieldType = "boolean"
    }
}

elsTerminaBoss.fieldOrder = {
    "x",
    "y",
    "phase",
    "fromCutscene",
    "hardMode"
}

function elsTerminaBoss.selection(room, entity)
    return utils.rectangle(entity.x - 40, entity.y - 60, 80, 120)
end

return elsTerminaBoss
